nuxeo-features/nuxeo-platform-imaging/nuxeo-platform-imaging-core/src/main/java/org/nuxeo/ecm/platform/picture/listener/PictureViewsGenerationListener.java

```diff
@@ -56,10 +58,12 @@ public void handleEvent(Event event) {
         DocumentEventContext docCtx = (DocumentEventContext) ctx;
         DocumentModel doc = docCtx.getSourceDocument();
         if (doc.hasFacet(PICTURE_FACET) && !doc.isProxy()) {
-            PictureViewsGenerationWork work = new PictureViewsGenerationWork(doc.getRepositoryName(), doc.getId(),
-                    "file:content");
-            WorkManager workManager = Framework.getService(WorkManager.class);
-            workManager.schedule(work, WorkManager.Scheduling.IF_NOT_SCHEDULED, true);
+            String query = "SELECT * FROM Document WHERE ecm:uuid='" + doc.getId() + "'";
+            BulkService service = Framework.getService(BulkService.class);
+            String username = ctx.getPrincipal().getName();
+            service.submit(new BulkCommand.Builder(ACTION_NAME, query, username)
+                    .param(PARAM_XPATH, "file:content")
+                    .build());
```

Comment:
Is it worth using a bulk action for a single document or does it just simplify to always use the `recomputeViews` action and deprecate the work?

Reply:
That was my point